tools/sdk-generation-pipeline/packages/sdk-generation-cli/src/utils/autorestConfigExtractorUtils.ts

```diff
@@ -0,0 +1,39 @@
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import winston from 'winston';
+
+export function extractAutorestConfigs(autorestConfigFilePath: string, sdkRepo: string, logger: winston.Logger): string | undefined {
+    if (!fs.existsSync(autorestConfigFilePath)) {
+        return undefined;
+    }
+    const autorestConfigFileContent = fs.readFileSync(autorestConfigFilePath, 'utf-8');
+    try {
+        let autorestConfigs: string[] = autorestConfigFileContent.split(/#+ *azure-sdk-for-/);
+        if (autorestConfigs.length < 2) {
+            throw new Error(`Parse autorest config file failed.`);
+        }
+        autorestConfigs = autorestConfigs.slice(1);
+        if (!path.basename(sdkRepo).startsWith('azure-sdk-for-')) {
+            if (autorestConfigs.length > 1) {
+                logger.warn(`Docker is running pipeline, but get autorest config for more than 1 language of sdk. So only get the first autorest config`);
+            }
+            sdkRepo = 'azure-sdk-for-';
+        }
+        for (const autorestConfig of autorestConfigs) {
+            let autorestFullConfig = `# azure-sdk-for-${autorestConfig}`;
+            if (autorestFullConfig.startsWith(`# ${path.basename(sdkRepo)}`)) {
```

Comment:
if you want to support that the pipeline is to generate more than one sdk, and the autorestConfig is written for more than one sdk, and then retrieve the first one. It better to write as following:
if (!path.basename(sdkRepo).startsWith('azure-sdk-for-')) {
ogger.warn(`Docker is running pipeline, but get autorest config for more than 1 language of sdk. So only get the first autorest config`);
  return `# azure-sdk-for-${autorestConfigs[0]}`;
} else {
  for (const autorestConfig of autorestConfigs) {
              let autorestFullConfig = `# azure-sdk-for-${autorestConfig}`;
              if (autorestFullConfig.startsWith(`# ${path.basename(sdkRepo)}`)) {
                  logger.info(`Find autorest config for ${path.basename(sdkRepo)} in ${autorestConfigFilePath}: \n${autorestFullConfig}`);
                  if (os.EOL == '\n') {
                      autorestFullConfig = autorestFullConfig.replace(/\n/g, '\r\n');
                  }
                  return autorestFullConfig;
              }
          }
}

Reply:
looks good. changed